File: 5-longest-palindromic-substring/longest-palindromic-substring.java
class Solution {
    public String longestPalindrome(String s) {
        StringBuilder res = new StringBuilder();
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            StringBuilder subString1 = new StringBuilder();
            StringBuilder subString2 = new StringBuilder();

            // check for odd palindrome
            int left = i - 1, right = i + 1;
            while (left >= 0 && right < s.length() && (s.charAt(left) == s.charAt(right))) {
                left--;
                right++;
            }

            for(int k = left + 1; k < right; k++) {
                subString1.append(s.charAt(k));
            }


            //check for even palindrome
            if (i + 1 < s.length() && s.charAt(i) == s.charAt(i + 1)) {
                left = i - 1;
                right = i + 2;
                while (left >= 0 && right < s.length() && (s.charAt(left) == s.charAt(right))) {
                    left--;
                    right++;
                }
                for(int k = left + 1; k < right; k++) {
                    subString2.append(s.charAt(k));
                }
            } else {
                subString2.append(c);
            }

            if (res.length() < subString1.length()) {
                res = subString1;
            }
            if (res.length() < subString2.length()) {
                res = subString2;
            }
        }
        return res.toString();
    }
}

File: 5-longest-palindromic-substring/longest-palindromic-substring.py
class Solution(object):
    def longestPalindrome(self, s):
        leftIndex, rightIndex = 0,0
        max_length = 0
        for i in range(len(s)):

            # check for odd palindrome
            left, right = i - 1, i + 1
            while left >= 0 and right < len(s) and s[left] == s[right]:
                left -= 1
                right += 1
            
            if right - left - 1 > max_length:
                max_length = right - left - 1
                leftIndex = left + 1
                rightIndex = right - 1
            
            # check for even palindrome
            if i + 1 < len(s) and s[i] == s[i+1]:
                left, right = i-1,i+2
                while left >= 0 and right < len(s) and s[left] == s[right]:
                    left -= 1
                    right += 1

                if right - left - 1 > max_length:
                    max_length = right - left - 1
                    leftIndex = left + 1
                    rightIndex = right - 1
        
        return s[leftIndex:rightIndex + 1]


        